src/generate.js

```diff
@@ -252,87 +265,79 @@ export const generateCSSRuleset = (
 
     // NOTE(emily): This mutates handledDeclarations.elements.
-    const prefixedDeclarations = prefixAll(handledDeclarations.elements);
-
-    const prefixedRules = flatten(
-        objectToPairs(prefixedDeclarations).map(([key, value]) => {
-            if (Array.isArray(value)) {
-                // inline-style-prefixer returns an array when there should be
-                // multiple rules for the same key. Here we flatten to multiple
-                // pairs with the same key.
-                return value.map(v => [key, v]);
-            }
-            return [[key, value]];
-        })
-    );
+    const prefixedElements = prefixAll(declarations.elements);
 
-    // Calculate the order that we want to each element in `prefixedRules` to
-    // be in, based on its index in the original key ordering.
-    const sortOrder = {};
-    for (let i = 0; i < handledDeclarations.keyOrder.length; i++) {
-        const key = handledDeclarations.keyOrder[i];
-        sortOrder[key] = i;
+    const elementNames = Object.keys(prefixedElements);
+    if (elementNames.length !== declarations.keyOrder.length) {
+        // There are some prefixed values, so we need to figure out how to sort
+        // them.
+        //
+        // Loop through prefixedElements, looking for anything that is not in
+        // sortOrder, which means it was added by prefixAll. This means that we
+        // need to figure out where it should appear in the sortOrder.
+        for (let i = 0; i < elementNames.length; i++) {
+            if (!originalElements.hasOwnProperty(elementNames[i])) {
+                // This element is not in the sortOrder, which means it is a prefixed
+                // value that was added by prefixAll. Let's try to figure out where it
+                // goes.
+                let originalStyle;
+                if (elementNames[i][0] === 'W') {
+                    // This is a Webkit-prefixed style, like "WebkitTransition". Let's
+                    // find its original style's sort order.
+                    originalStyle = elementNames[i][6].toLowerCase() + elementNames[i].slice(7);
+                } else if (elementNames[i][1] === 'o') {
+                    // This is a Moz-prefixed style, like "MozTransition". We check
+                    // the second character to avoid colliding with Ms-prefixed
+                    // styles. Let's find its original style's sort order.
+                    originalStyle = elementNames[i][3].toLowerCase() + elementNames[i].slice(4);
+                } else if (elementNames[i][1] === 's') {
+                    // This is a Ms-prefixed style, like "MsTransition". We check
+                    // the second character to avoid colliding with Moz-prefixed
+                    // styles. Let's find its original style's sort order.
+                    originalStyle = elementNames[i][2].toLowerCase() + elementNames[i].slice(3);
+                }
```

Comment:
Looks like coverage is failing branch coverage for the `else` case here. Any thoughts on how to fix this?